Source/msg.cpp

```diff
@@ -1094,17 +1096,18 @@ DWORD OnRangedAttackPlayer(TCmd *pCmd, Player &player)
 
 DWORD OnSpellMonster(TCmd *pCmd, Player &player)
 {
-	auto *p = (TCmdParam3 *)pCmd;
+	auto *p = (TCmdParam4 *)pCmd;
 
 	if (gbBufferMsgs != 1 && currlevel == player.plrlevel) {
 		auto spell = static_cast<spell_id>(p->wParam2);
+		auto spellType = static_cast<spell_type>(p->wParam3);

```

Comment:
I think pulling out the parameters at the start of the process like this gives us a good place to perform input validation in preparation for https://github.com/diasurgical/devilutionX/issues/2641 and https://github.com/diasurgical/devilutionX/issues/2707 :+1: